core/src/main/scala/com/typesafe/tools/mima/lib/analyze/method/MethodChecker.scala

```diff
@@ -34,15 +35,24 @@ private[analyze] abstract class BaseMethodChecker extends Checker[MethodInfo, Cl
     }
   }
 
-  private def methSigCheck(oldmeth: MethodInfo, newMeth: MethodInfo): Boolean = {
-    oldmeth.signature == newMeth.signature || (
-      newMeth.bytecodeName == MemberInfo.ConstructorName && newMeth.signature.isEmpty
-    )
-  }
-
   private def uniques(methods: List[MethodInfo]): List[MethodInfo] =
     methods.groupBy(_.parametersDesc).values.map(_.head).toList
 }
+private[analyze] object BaseMethodChecker {
+  def hasMatchingDescriptorAndSignature(oldMethod: MethodInfo, newMethod: MethodInfo): Boolean =
+    oldMethod.descriptor == newMethod.descriptor && hasMatchingSignature(oldMethod.signature, newMethod.signature, newMethod.bytecodeName)
+
+  // Assumes it is already checked that the descriptors match
+  def hasMatchingSignature(oldSignature: String, newSignature: String, bytecodeName: String): Boolean =
+  oldSignature == newSignature ||
+    // Special case for https://github.com/scala/scala/pull/7975:
+    (bytecodeName == MemberInfo.ConstructorName &&
+      (newSignature.isEmpty ||
+        // The dropped character is the leading '('
+        oldSignature.endsWith(newSignature.tail)
```

Comment:
> It's now 1 char - the leading `(` of the new signature (which may have dropped the first parameter compared to the old).

If that's the case can it be `oldSignature == newSignature.tail`?  Sorry, I'm not groking the change in the compiler exactly, so I'm just trying to make it least likely to false negative.

Reply:
The change is that the old Scala version produced an extra parameter at the beginning, where the new (correctly) does not.

So, like in the test, we want to allow having `(`(firstparameter)(otherparameters`);`

Reply:
OK, I get it now.  Thanks.

Reply:
You got it :+1: , thanks for nudging me to make this more clear in the code (and add a test), that both makes it more readable and we found a bug that way ;).